### PR TITLE
Add support for extracting CoSigV1 timestamps from signatures

### PR DESCRIPTION
**Pull request by AlCutter**

### CODE REVIEW

File: note/note_cosigv1.go [hunk 1]
```diff
@@ -152,6 +166,7 @@ var (
 	errSignerAlg   = errors.New("unknown signer algorithm")
 	errVerifierID  = errors.New("malformed verifier id")
 	errVerifierAlg = errors.New("unknown verifier algorithm")
+	errMalformed   = errors.New("malformed signature")
```
Comment (roger2hk) [inline note/note_cosigv1.go:169]: nit: `errMalformedSig` or `errSignature` to align with other errors

File: note/note_cosigv1.go [hunk 2]
```diff
@@ -107,6 +107,20 @@ func NewVerifierForCosignatureV1(vkey string) (note.Verifier, error) {
 	return v, nil
 }
 
+// CoSigV1Timestamp extracts the embedded timestamp from a CoSigV1 signature.
+func CoSigV1Timestamp(s note.Signature) (time.Time, error) {
+	r, err := base64.StdEncoding.DecodeString(s.Base64)
+	if err != nil {
+		return time.UnixMilli(0), errMalformed
+	}
+	if len(r) != 4+8+ed25519.SignatureSize {
```
Comment (roger2hk) [inline note/note_cosigv1.go:116]: Could the `4` and `8` be constants to tell what they are?

**Review by roger2hk — APPROVED** [review]

File: note/note_cosigv1.go (hunk 2 above)
Comment (AlCutter) [reply note/note_cosigv1.go]: I was following the style elsewhere in this file, but I've added two consts.

**Comment by AlCutter** [comment]
Ta, PTAL

**Pull request merged**